cmd/ui/src/rendering/programs/edge.selfArrowHead.ts

```diff
@@ -0,0 +1,118 @@
+// Copyright 2024 Specter Ops, Inc.
+//
+// Licensed under the Apache License, Version 2.0
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+/**
+ * Sigma.js WebGL Renderer Arrow Program
+ * ======================================
+ *
+ * Program rendering direction arrows as a simple triangle.
+ * @module
+ */
+import CurvedEdgeArrowHeadProgram from './edge.curvedArrowHead';
+import { NodeDisplayData } from 'sigma/types';
+import { floatColor } from 'sigma/utils';
+import { Attributes } from 'graphology-types';
+import { bezier } from 'src/rendering/utils/bezier';
+import { getNodeRadius } from 'src/rendering/utils/utils';
+import { getControlPointsFromGroupSize } from './edge.self';
+
+const POINTS = 3,
+    ATTRIBUTES = 9,
+    STRIDE = POINTS * ATTRIBUTES;
+
+export default class SelfEdgeArrowHeadProgram extends CurvedEdgeArrowHeadProgram {
+    constructor(gl: WebGLRenderingContext) {
+        super(gl);
+    }
+
+    process(
+        sourceData: NodeDisplayData,
+        targetData: NodeDisplayData,
+        data: Attributes,
+        hidden: boolean,
+        offset: number
+    ): void {
+        if (hidden) {
+            for (let i = offset * STRIDE, l = i + STRIDE; i < l; i++) this.array[i] = 0;
+            return;
+        }
+
+        const inverseSqrtZoomRatio = data.inverseSqrtZoomRatio || 1;
+        const start = { x: sourceData.x, y: sourceData.y };
+        const thickness = data.size || 1;
+        const radius = getNodeRadius(targetData.highlighted, inverseSqrtZoomRatio, targetData.size);
+        const color = floatColor(data.color);
+        const { control2, control3 } = getControlPointsFromGroupSize(
+            data.groupPosition,
+            data.framedGraphNodeRadius * 3,
+            start,
+            false,
+            false
+        );
+
+        const curveCircleIntersectionApproximation = bezier.getCoordinatesAlongCubicBezier(
+            start,
+            control2,
+            control3,
+            start,
+            0.8975
```

Comment:
It might be good to pull this number into a constant to give a little more context to how we are doing this approximation